Feature/Xibo RDM Connector

mgbaybay opened the pull request:
## Changes
Xibo RDM Connector Implementation

mgbaybay committed:
RDM Devices: Connector setup

mgbaybay committed:
RDM Devices: Updated filter settings

mgbaybay committed:
RDM Connector: Add manual linking

mgbaybay committed:
RDM Connector: Add application migration

mgbaybay committed:
RDM Connector: Added comments

PeterMis commented on views/xibo-rdm-connector-form-javascript.twig:
```diff
@@ -0,0 +1,501 @@
+                                      { optionid: "Sony Bravia", option: "Sony Bravia" },
+                                      { optionid: "DSDevices", option: "DSDevices" },
+                                      { optionid: "Windows", option: "Windows" },
+                                      { optionid: "Chrome OS", option: "Chrome OS" },
```
It should be `ChromeOS` please

I'm not entirely sure if we should list ChromeOS/Windows as only Sony Bravia and to some extent DSDevices will be supported for now - but that can be changed later on

PeterMis commented on views/xibo-rdm-connector-form-javascript.twig:
```diff
@@ -0,0 +1,501 @@
+                                    { optionid: "Sony Bravia", option: "Sony Bravia" },
+                                    { optionid: "DSDevices", option: "DSDevices" },
+                                    { optionid: "Windows", option: "Windows" },
+                                    { optionid: "Chrome Os", option: "Chrome OS" },
```
ChromeOS

PeterMis commented on views/xibo-rdm-connector-form-javascript.twig:
```diff
@@ -0,0 +1,501 @@
+                                { optionid: "Sony Bravia", option: "Sony Bravia" },
+                                { optionid: "DSDevices", option: "DSDevices" },
+                                { optionid: "Windows", option: "Windows" },
+                                { optionid: "Chrome OS", option: "Chrome OS" },
```
ChromeOS

PeterMis commented on lib/Connector/XiboRdmConnector.php:
```diff
@@ -0,0 +1,378 @@
+     */
+    private function getServiceUrl(): string
+    {
+        return $this->getSetting('serviceUrl', 'http://xibosignage.com');
```
https perhaps?

PeterMis reviewed (changes requested):
Looks good, just a couple of minor comments to address

Since the phpcs errors are on the newly added code, we should fix them all please
( make sure you have phpcs configured in your IDE as well, as all of those should be highlighted there )

mgbaybay committed:
RDM Connector: Fixed phpcs

PeterMis reviewed (dismissed):
Looks good for QA I think 👍

mgbaybay committed:
RDM Connector: Fixed issue with adding displays

mgbaybay committed:
RDM Connector: Update table filter

mgbaybay committed:
RDM Devices: Display Profile Migrations

mgbaybay committed:
RDM Devices: Display Commands Setup

mgbaybay committed:
RDM Devices: Add DSDevices command type validation

mgbaybay committed:
Merge branch 'develop' into feature/xibo_rdm_connector

maurofmferrao committed:
Initial work for RDM arrow linking
relates to xibosignageltd/xibo-private#991

mgbaybay committed:
RDM Devices: Implement arrow linking

mgbaybay committed:
RDM Devices: Manual connection fix

mgbaybay committed:
RDM Devices: Update screenshot functionality for sony displays

maurofmferrao committed:
Multiple fixes for RDM connector
relates to xibosignageltd/xibo-private#991

- Prevent selecting last dummy row
- Wrongly name Display text to Device

mgbaybay committed:
RDM Devices: Removed install apk command

mgbaybay committed:
RDM Devices: Minor BE fixes

mgbaybay committed:
RDM Devices: Fix concatenation error

mgbaybay committed:
RDM Devices: Revert display profile changes

PeterMis reviewed (approved):
Approving for QA.
We will need to work out the outstanding items before we actually merge though please :)
